### ISSUE TITLE
Installation failing for CmdStan v. 2.22.0

### ISSUE THREAD

**Issue by justinbois** [stan-dev/cmdstanpy#200]
#### Summary:
Installation on MacOS of CmdStan v. 2.22.0 fails.

#### Description:
I attempted to install CmdStan v. 2.22.0 by doing the following at the command line:

`python -c "import cmdstanpy; cmdstanpy.install_cmdstan()";`

The `~/.cmdstanpy` directory did not exist prior to running this command.

The results are:
```
$ python -c "import cmdstanpy; cmdstanpy.install_cmdstan()";
INFO:cmdstanpy:CmdStan version: 2.22.0
INFO:cmdstanpy:Install directory: /Users/bois/.cmdstanpy
INFO:cmdstanpy:Downloading CmdStan version 2.22.0
INFO:cmdstanpy:Download successful, file: /var/folders/tj/4wnqnhc13cl70g85k6w4kbh80000gn/T/tmpoxhtudl7
INFO:cmdstanpy:Unpacked download as cmdstan-2.22.0
INFO:cmdstanpy:Building cmdstan-2.22.0 binaries
INFO:cmdstanpy:Command "make build" failed
INFO:cmdstanpy:/bin/sh: -c: line 0: syntax error near unexpected token `)'
INFO:cmdstanpy:/bin/sh: -c: line 0: `cp bin/mac-stanc) bin/stanc'
INFO:cmdstanpy:make: *** [bin/stanc] Error 2
WARNING:cmdstanpy:CmdStan installation failed
deleting tmpfiles dir: /var/folders/tj/4wnqnhc13cl70g85k6w4kbh80000gn/T/tmpkeqfp8w6
done

``` 

However installation of v. 2.21.0 is successful.

```
$ python -c "import cmdstanpy; cmdstanpy.install_cmdstan(version='2.21.0')";
INFO:cmdstanpy:CmdStan version: 2.21.0
INFO:cmdstanpy:Install directory: /Users/bois/.cmdstanpy
INFO:cmdstanpy:Downloading CmdStan version 2.21.0
INFO:cmdstanpy:Download successful, file: /var/folders/tj/4wnqnhc13cl70g85k6w4kbh80000gn/T/tmp1s17snp7
INFO:cmdstanpy:Unpacked download as cmdstan-2.21.0
INFO:cmdstanpy:Building cmdstan-2.21.0 binaries
INFO:cmdstanpy:Test model compilation
INFO:cmdstanpy:Installed cmdstan-2.21.0
deleting tmpfiles dir: /var/folders/tj/4wnqnhc13cl70g85k6w4kbh80000gn/T/tmpruubvn6c
done

```

#### Additional Information:
macos: Version 10.15.2
XCode: Version 11.3.1 (11C504)

#### Current Version:
cmdstanpy: 0.8.0

**Comment by ahartikainen** [comment]
Hi, thanks for the info. Do you know what happens if you try to install CmdStan 2.22 manually?

**Comment by justinbois** [comment]
The results from trying to install CmdStan2.22 manually on the same Mac as above are shown below.

It is also worth noting that I tried the installation using CmdStanPy on an AWS instance (Amazon Linux 2 AMI (HVM), SSD Volume Type - ami-04590e7389a6e577c (64-bit x86) / ami-0dfa0289aa742513b (64-bit Arm)). I again used `python -c "import cmdstanpy; cmdstanpy.install_cmdstan()";`. I got the same failure message as I did on my Mac.

#### Messages for installation via CmdStanPy on AWS:
```
INFO:cmdstanpy:CmdStan version: 2.22.0
INFO:cmdstanpy:Install directory: /home/ec2-user/.cmdstanpy
INFO:cmdstanpy:Downloading CmdStan version 2.22.0
INFO:cmdstanpy:Download successful, file: /tmp/tmpukppu69o
INFO:cmdstanpy:Unpacked download as cmdstan-2.22.0
INFO:cmdstanpy:Building cmdstan-2.22.0 binaries
INFO:cmdstanpy:Command "make build" failed
INFO:cmdstanpy:/bin/sh: -c: line 0: syntax error near unexpected token `)'
INFO:cmdstanpy:/bin/sh: -c: line 0: `cp bin/linux-stanc) bin/stanc'
INFO:cmdstanpy:make: *** [bin/stanc] Error 1
WARNING:cmdstanpy:CmdStan installation failed
deleting tmpfiles dir: /tmp/tmpe9czv0qv
done
```


#### CmdStan installation output

```
$ make build -j1
clang++ -std=c++1y -Wno-unknown-warning-option -Wno-tautological-compare -Wno-sign-compare -D_REENTRANT       -I stan/lib/stan_math/lib/tbb_2019_U8/include -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.3 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_4.1.0/include    -DBOOST_DISABLE_ASSERTS      -c -MT src/cmdstan/stanc.o -MM -E -MG -MP -MF src/cmdstan/stanc.d src/cmdstan/stanc.cpp
clang++ -std=c++1y -Wno-unknown-warning-option -Wno-tautological-compare -Wno-sign-compare -D_REENTRANT       -I stan/lib/stan_math/lib/tbb_2019_U8/include -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.3 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_4.1.0/include    -DBOOST_DISABLE_ASSERTS      -c -MT bin/cmdstan/lang/ast_def.o -MT stan/src/stan/lang/ast_def.d -MM -E -MG -MP -MF stan/src/stan/lang/ast_def.d stan/src/stan/lang/ast_def.cpp
clang++ -std=c++1y -Wno-unknown-warning-option -Wno-tautological-compare -Wno-sign-compare -D_REENTRANT       -I stan/lib/stan_math/lib/tbb_2019_U8/include -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.3 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_4.1.0/include    -DBOOST_DISABLE_ASSERTS      -c -MT bin/cmdstan/lang/grammars/semantic_actions_def.o -MT stan/src/stan/lang/grammars/semantic_actions_def.d -MM -E -MG -MP -MF stan/src/stan/lang/grammars/semantic_actions_def.d stan/src/stan/lang/grammars/semantic_actions_def.cpp
clang++ -std=c++1y -Wno-unknown-warning-option -Wno-tautological-compare -Wno-sign-compare -D_REENTRANT       -I stan/lib/stan_math/lib/tbb_2019_U8/include -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.3 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_4.1.0/include    -DBOOST_DISABLE_ASSERTS      -c -MT bin/cmdstan/lang/grammars/statement_grammar_inst.o -MT stan/src/stan/lang/grammars/statement_grammar_inst.d -MM -E -MG -MP -MF stan/src/stan/lang/grammars/statement_grammar_inst.d stan/src/stan/lang/grammars/statement_grammar_inst.cpp
clang++ -std=c++1y -Wno-unknown-warning-option -Wno-tautological-compare -Wno-sign-compare -D_REENTRANT       -I stan/lib/stan_math/lib/tbb_2019_U8/include -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.3 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_4.1.0/include    -DBOOST_DISABLE_ASSERTS      -c -MT bin/cmdstan/lang/grammars/bare_type_grammar_inst.o -MT stan/src/stan/lang/grammars/bare_type_grammar_inst.d -MM -E -MG -MP -MF stan/src/stan/lang/grammars/bare_type_grammar_inst.d stan/src/stan/lang/grammars/bare_type_grammar_inst.cpp
clang++ -std=c++1y -Wno-unknown-warning-option -Wno-tautological-compare -Wno-sign-compare -D_REENTRANT       -I stan/lib/stan_math/lib/tbb_2019_U8/include -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.3 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_4.1.0/include    -DBOOST_DISABLE_ASSERTS      -c -MT bin/cmdstan/lang/grammars/indexes_grammar_inst.o -MT stan/src/stan/lang/grammars/indexes_grammar_inst.d -MM -E -MG -MP -MF stan/src/stan/lang/grammars/indexes_grammar_inst.d stan/src/stan/lang/grammars/indexes_grammar_inst.cpp
clang++ -std=c++1y -Wno-unknown-warning-option -Wno-tautological-compare -Wno-sign-compare -D_REENTRANT       -I stan/lib/stan_math/lib/tbb_2019_U8/include -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.3 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_4.1.0/include    -DBOOST_DISABLE_ASSERTS      -c -MT bin/cmdstan/lang/grammars/local_var_decls_grammar_inst.o -MT stan/src/stan/lang/grammars/local_var_decls_grammar_inst.d -MM -E -MG -MP -MF stan/src/stan/lang/grammars/local_var_decls_grammar_inst.d stan/src/stan/lang/grammars/local_var_decls_grammar_inst.cpp
clang++ -std=c++1y -Wno-unknown-warning-option -Wno-tautological-compare -Wno-sign-compare -D_REENTRANT       -I stan/lib/stan_math/lib/tbb_2019_U8/include -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.3 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_4.1.0/include    -DBOOST_DISABLE_ASSERTS      -c -MT bin/cmdstan/lang/grammars/expression07_grammar_inst.o -MT stan/src/stan/lang/grammars/expression07_grammar_inst.d -MM -E -MG -MP -MF stan/src/stan/lang/grammars/expression07_grammar_inst.d stan/src/stan/lang/grammars/expression07_grammar_inst.cpp
clang++ -std=c++1y -Wno-unknown-warning-option -Wno-tautological-compare -Wno-sign-compare -D_REENTRANT       -I stan/lib/stan_math/lib/tbb_2019_U8/include -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.3 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_4.1.0/include    -DBOOST_DISABLE_ASSERTS      -c -MT bin/cmdstan/lang/grammars/statement_2_grammar_inst.o -MT stan/src/stan/lang/grammars/statement_2_grammar_inst.d -MM -E -MG -MP -MF stan/src/stan/lang/grammars/statement_2_grammar_inst.d stan/src/stan/lang/grammars/statement_2_grammar_inst.cpp
clang++ -std=c++1y -Wno-unknown-warning-option -Wno-tautological-compare -Wno-sign-compare -D_REENTRANT       -I stan/lib/stan_math/lib/tbb_2019_U8/include -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.3 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_4.1.0/include    -DBOOST_DISABLE_ASSERTS      -c -MT bin/cmdstan/lang/grammars/block_var_decls_grammar_inst.o -MT stan/src/stan/lang/grammars/block_var_decls_grammar_inst.d -MM -E -MG -MP -MF stan/src/stan/lang/grammars/block_var_decls_grammar_inst.d stan/src/stan/lang/grammars/block_var_decls_grammar_inst.cpp
clang++ -std=c++1y -Wno-unknown-warning-option -Wno-tautological-compare -Wno-sign-compare -D_REENTRANT       -I stan/lib/stan_math/lib/tbb_2019_U8/include -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.3 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_4.1.0/include    -DBOOST_DISABLE_ASSERTS      -c -MT bin/cmdstan/lang/grammars/term_grammar_inst.o -MT stan/src/stan/lang/grammars/term_grammar_inst.d -MM -E -MG -MP -MF stan/src/stan/lang/grammars/term_grammar_inst.d stan/src/stan/lang/grammars/term_grammar_inst.cpp
clang++ -std=c++1y -Wno-unknown-warning-option -Wno-tautological-compare -Wno-sign-compare -D_REENTRANT       -I stan/lib/stan_math/lib/tbb_2019_U8/include -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.3 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_4.1.0/include    -DBOOST_DISABLE_ASSERTS      -c -MT bin/cmdstan/lang/grammars/whitespace_grammar_inst.o -MT stan/src/stan/lang/grammars/whitespace_grammar_inst.d -MM -E -MG -MP -MF stan/src/stan/lang/grammars/whitespace_grammar_inst.d stan/src/stan/lang/grammars/whitespace_grammar_inst.cpp
clang++ -std=c++1y -Wno-unknown-warning-option -Wno-tautological-compare -Wno-sign-compare -D_REENTRANT       -I stan/lib/stan_math/lib/tbb_2019_U8/include -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.3 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_4.1.0/include    -DBOOST_DISABLE_ASSERTS      -c -MT bin/cmdstan/lang/grammars/functions_grammar_inst.o -MT stan/src/stan/lang/grammars/functions_grammar_inst.d -MM -E -MG -MP -MF stan/src/stan/lang/grammars/functions_grammar_inst.d stan/src/stan/lang/grammars/functions_grammar_inst.cpp
clang++ -std=c++1y -Wno-unknown-warning-option -Wno-tautological-compare -Wno-sign-compare -D_REENTRANT       -I stan/lib/stan_math/lib/tbb_2019_U8/include -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.3 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_4.1.0/include    -DBOOST_DISABLE_ASSERTS      -c -MT bin/cmdstan/lang/grammars/program_grammar_inst.o -MT stan/src/stan/lang/grammars/program_grammar_inst.d -MM -E -MG -MP -MF stan/src/stan/lang/grammars/program_grammar_inst.d stan/src/stan/lang/grammars/program_grammar_inst.cpp
clang++ -std=c++1y -Wno-unknown-warning-option -Wno-tautological-compare -Wno-sign-compare -D_REENTRANT       -I stan/lib/stan_math/lib/tbb_2019_U8/include -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.3 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_4.1.0/include    -DBOOST_DISABLE_ASSERTS      -c -MT bin/cmdstan/lang/grammars/expression_grammar_inst.o -MT stan/src/stan/lang/grammars/expression_grammar_inst.d -MM -E -MG -MP -MF stan/src/stan/lang/grammars/expression_grammar_inst.d stan/src/stan/lang/grammars/expression_grammar_inst.cpp
cp bin/mac-stanc) bin/stanc
/bin/sh: -c: line 0: syntax error near unexpected token `)'
/bin/sh: -c: line 0: `cp bin/mac-stanc) bin/stanc'
make: *** [bin/stanc] Error 2
```

**Comment by ahartikainen** [comment]
See https://github.com/stan-dev/cmdstan/issues/804

**Comment by MooersLab** [comment]
I ran into the same problem.
I issued the command  python3.8 -c "import cmdstanpy; cmdstanpy.install_cmdstan(version='2.22.0')"
I got the following error message:

```bash
NFO:cmdstanpy:CmdStan version: 2.22.0
INFO:cmdstanpy:Install directory: /Users/blaine/.cmdstanpy
INFO:cmdstanpy:Downloading CmdStan version 2.22.0
INFO:cmdstanpy:Download successful, file: /var/folders/mv/1w6nby494g5b3vjf661ktyfr0000gn/T/tmpf3yiw4y8
INFO:cmdstanpy:Unpacked download as cmdstan-2.22.0
INFO:cmdstanpy:Building cmdstan-2.22.0 binaries
INFO:cmdstanpy:Command "make build" failed
INFO:cmdstanpy:/bin/sh: -c: line 0: syntax error near unexpected token `)'
INFO:cmdstanpy:/bin/sh: -c: line 0: `cp bin/mac-stanc) bin/stanc'
INFO:cmdstanpy:make: *** [bin/stanc] Error 2
WARNING:cmdstanpy:CmdStan installation failed
deleting tmpfiles dir: /var/folders/mv/1w6nby494g5b3vjf661ktyfr0000gn/T/tmprwr0m3k_
done
```
In .cmdstanpy, I found the bin subdirectory with mac_stanc present.  I copied it to stanc. I then run `make build`.

**Comment by mitzimorris** [comment]
CmdStan 2.22.0 had a bug in the makefile; it will be fixed in 2.22.1 release - which, we hope, will be out soon.

**Comment by mitzimorris** [comment]
cmdstan 2.22.1 is out, `cmdstanpy.install_cmdstan()` works on my machine - closing this
